activerecord/lib/active_record/database_configurations.rb

```diff
@@ -205,12 +205,14 @@ def build_configuration_sentence
 
         configs.group_by(&:env_name).map do |env, config|
           names = config.map(&:name)
+          next if env == "default"
```

Comment:
This isn't correct. While the generated on is "default" we don't know that it's actually going to be called default. We have no way of knowing a config isn't valid so we have to include all of them in the list.

Reply:
It's entirely possible someone does have an env named default and it's valid, so skipping it could break applications.